### PR TITLE
Remove use of `getPointerElementType` in `gen/arrays.cpp`

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

File: gen/arrays.cpp [hunk 1]
```diff
@@ -853,6 +833,27 @@ DSliceValue *DtoCatAssignArray(const Loc &loc, DValue *arr, Expression *exp) {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+static LLValue *DtoSlicePtr(DValue *dval) {
```
Comment (thewilsonator) [inline gen/arrays.cpp:836]: I moved this closer to its only user

**Review by thewilsonator — COMMENTED** [review]
Yay finally! This is ready for review.

File: gen/arrays.cpp (hunk 1 above)
Comment (JohanEngelen) [reply gen/arrays.cpp:836]: it's now really hard to see what you changed, vs simply moved.

Comment (thewilsonator) [reply gen/arrays.cpp:836]: Hmm, I must have left that in when trying to figure out the return type.
I can revert that back if desired, but the changes are:

* `s/type/i8arrty` and moved its definition to the point of use, past the early return, and 
* split off the `toElem(e)` into a separate function.

**Pull request merged**